### ISSUE TITLE
[Android] Broken Image with AspectFill

### ISSUE THREAD

**Issue by rredoh** [xamarin/Xamarin.Forms#4133]
### Description
I don't know how to explain, but I'll give some screenshot and some ViewCell

### Steps to Reproduce
1. Enable Fast Renderers on the control gallery project
2. Run control gallery
3. Go to Image Gallery
4. Set to Aspect Fill

![image](https://user-images.githubusercontent.com/5375137/47199132-50f4a700-d32d-11e8-9edb-76f09bfdc650.png)

### Steps to Reproduce Original

1. Add a ListView
2. Make a ViewCell
3. Add Image with AspectFill
4. Broken

- Version with issue: I'm now at Xamarin Forms 3.3.0.912
- Last known good version: 3.2
- IDE: Visual Studio 15.8.7
- Platform Target Frameworks:
  - Android: 8.1
- Android Support Library Version: 27.0.2.1

### Screenshots
![image aspectfill](https://user-images.githubusercontent.com/16044363/47198359-ecb9fd80-d395-11e8-9ca7-32802efd4cc9.jpg)
### Some ViewCell code
```
<?xml version="1.0" encoding="UTF-8"?>
<ViewCell xmlns="http://xamarin.com/schemas/2014/forms" 
          xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
          x:Class="Assetku.ViewCells.Item_ViewCell">
    <Frame Padding="0" Margin="10" x:Name="frame">
        <Grid HorizontalOptions="FillAndExpand"
              VerticalOptions="FillAndExpand"
              Padding="10">
            <Grid.ColumnDefinitions>
                <ColumnDefinition Width="100"/>
                <ColumnDefinition Width="Auto"/>
                <ColumnDefinition Width="10"/>
                <ColumnDefinition Width="*"/>
            </Grid.ColumnDefinitions>
            <Grid.RowDefinitions>
                <RowDefinition Height="20"/>
                <RowDefinition Height="Auto"/>
                <RowDefinition Height="20"/>
                <RowDefinition Height="20"/>
                <RowDefinition Height="20"/>
            </Grid.RowDefinitions>
            <Image Grid.Column="0" Grid.Row="0" Grid.RowSpan="5"
                                 Margin="0,0,10,0" Source="{Binding IMAGE_LINK}"
                                 Aspect="AspectFill"/>

            <Label Text="{Binding NAME}" FontSize="Medium" FontFamily="{StaticResource BoldFont}"
                   LineBreakMode="TailTruncation"
                   Grid.Column="1" Grid.Row="0" Grid.ColumnSpan="3"/>

            <Label Text="{Binding ASKING_PRICE, StringFormat='{0:C}'}"
                   FontSize="Medium"
                   TextColor="#FF6F00"
                   LineBreakMode="TailTruncation"
                   Grid.Column="1" Grid.Row="1" Grid.ColumnSpan="3"/>
            
            <Label Text="Kategori"
                   Grid.Column="1" Grid.Row="2"/>
            <Label Text="Bunga"
                   Grid.Column="1" Grid.Row="3"/>
            <Label Text="Durasi"
                   Grid.Column="1" Grid.Row="4"/>

            <Label Text=":"
                   Grid.Column="2" Grid.Row="2"/>
            <Label Text=":"
                   Grid.Column="2" Grid.Row="3"/>
            <Label Text=":"
                   Grid.Column="2" Grid.Row="4"/>

            <Label Text="{Binding CATEGORY}"
                   LineBreakMode="TailTruncation"
                   Grid.Column="3" Grid.Row="2"/>
            <Label Text="{Binding ASKING_INTEREST_RATE}"
                   LineBreakMode="TailTruncation"
                   Grid.Column="3" Grid.Row="3"/>
            <Label Text="{Binding DURATION}"
                   LineBreakMode="TailTruncation"
                   Grid.Column="3" Grid.Row="4"/>
            
        </Grid>
    </Frame>
</ViewCell>
```

**Comment by PureWeen** [comment]
I wonder if it's caused from this clipping change?
https://github.com/xamarin/Xamarin.Forms/commit/d6aba7938c226a3ba6d8c45f848046028d182fbd

**Comment by rredoh** [comment]
Here is a screenshot from the old version. The image should stay inside its cell like this :
![image aspectfill old](https://user-images.githubusercontent.com/16044363/47208865-2a7c4d80-d3b9-11e8-82ef-f74ec49c8703.jpg)

And this is the image I use. Just random picking an online image from google image search.
![gold image](https://fm.cnbc.com/applications/cnbc.com/resources/img/editorial/2017/10/02/104748034-GettyImages-90075023.530x298.jpg?v=1506995840)

By the way thanks for the fast response. :)
@hartez

**Comment by brittonbeckham** [comment]
I am also seeing this in my app when I place and `ImageView` in a `Grid`. Works as intended in `XF 3.2` stable, broken in `XF 3.3` stable and `XF 3.4-pre1`

**Comment by 100s** [comment]
We also have this issue in XF 3.3

**Comment by rredoh** [comment]
May I ask if the fix is included in Xamarin Forms 3.3.0.967583? Because it's still not working for me. Thanks
